gadget/update.go

```diff
@@ -326,6 +327,95 @@ func EnsureLayoutCompatibility(gadgetLayout *LaidOutVolume, diskLayout *OnDiskVo
 	return nil
 }
 
+// DiskVolumeDeviceTraitsForDevice takes a gadget volume and an expected
+// disk device path and builds up the traits for that device. It validates that
+// the specified volume can be laid out exactly to the disk device, returning an
+// error if the two are not compatible.
+func DiskVolumeDeviceTraitsForDevice(laidOutVolume *LaidOutVolume, dev string) (res DiskVolumeDeviceTraits, err error) {
+	vol := laidOutVolume.Volume
+
+	// get the disk layout for this device
+	diskLayout, err := OnDiskVolumeFromDevice(dev)
+	if err != nil {
+		return res, fmt.Errorf("cannot read %v partitions for candidate volume %s: %v", dev, vol.Name, err)
+	}
+
+	// ensure that the on disk and the laid out version are actually compatible
```

Comment:
```suggestion
	// ensure that the disk and the planned laid out version are actually compatible
```

Reply:
I went with something slightly different, but hopefully this is more clear now